### PR TITLE
zcash_client_sqlite: Prevent scan range fragmentation.

### LINKED ISSUE #922: zcash_client_sqlite: adjacent scan ranges are not being correctly unified
Low-ish priority.

Observed in the wallet state provided by @LukasKorba in the process of investigating #898:

```
sqlite> select * from scan_queue;
280000|490074|0
490074|2170000|50
2170000|2179110|10
2179110|2179210|10
2179210|2179310|10
2179310|2179410|10
2179410|2179510|10
2179510|2179610|10
2179610|2179710|10
2179710|2179810|10
2179810|2179910|10
2179910|2180010|10
2180010|2180110|10
2180110|2180210|10
2180210|2180310|10
2180310|2180410|10
2180410|2180510|10
2180510|2180610|10
2180610|2180710|10
2180710|2180810|10
2180810|2180910|10
2180910|2181010|10
2181010|2181110|10
2181110|2181210|10
2181210|2181310|10
2181310|2181410|10
2181410|2181510|10
2181510|2181610|10
2181610|2181710|10
2181710|2181810|10
2181810|2181910|10
2181910|2182010|10
2182010|2182110|10
2182110|2182210|10
2182210|2182310|10
2182310|2182410|10
2182410|2182510|10
2182510|2182610|10
2182610|2182710|10
2182710|2182810|10
2182810|2182910|10
2182910|2183010|10
2183010|2183110|10
2183110|2183210|10
2183210|2183310|10
2183310|2183410|10
2183410|2489003|50
```

It's not clear why all of those 100-block-wide scanned ranges have not been coalesced. It might be because when scanning the `ChainTip` range `2183410|2489003|50`, each successive 10-block range that is being scanned takes a piece of that range without affecting the previously scanned ranges. Since the prior range is not inspected, there's no opportunity to coalesce ranges. This isn't semantically incorrect, but it could lead to diminished performance of queries in mainnet wallets.

### PR DESCRIPTION
**Pull request by nuttycom**
Fixes #922

### CODE REVIEW

File: zcash_client_sqlite/src/wallet/scanning.rs [hunk 1]
```diff
@@ -455,7 +455,7 @@ pub(crate) fn replace_queue_entries(
             WHERE (
                 -- the start is contained within the range
```
Comment (str4d) [inline zcash_client_sqlite/src/wallet/scanning.rs:456]:
```suggestion
                -- the start is contained within or adjacent to the range
```

**Review by str4d — DISMISSED** [review]
utACK

**Review by str4d — APPROVED** [review]
utACK

File: zcash_client_sqlite/src/wallet/scanning.rs [hunk 2]
```diff
@@ -453,9 +453,9 @@ pub(crate) fn replace_queue_entries(
...
             FROM scan_queue
             WHERE (
-                -- the start is contained within the range
+                -- the start is contained within or adjacent to the range
                 :start >= block_range_start
-                AND :start < block_range_end
+                AND :start <= block_range_end
             )
             OR (
                 -- the end is contained within the range
```
Comment (daira) [inline zcash_client_sqlite/src/wallet/scanning.rs:461]: Is it intended that this merges with a range adjacent to the start but not adjacent to the end?

Comment (nuttycom) [reply zcash_client_sqlite/src/wallet/scanning.rs:461]: Ah... good point. I'll fix that too.

**Review by ebfull — APPROVED** [review]

**Pull request merged**